crates/bevy_audio/src/volume.rs

```diff
@@ -155,105 +155,195 @@ impl Volume {
+        if current_db == f32::NEG_INFINITY {
+            return if decibel_offset == f32::NEG_INFINITY {
+                Volume::SILENT
+            } else {
+                // When the current volume is silent, the offset becomes the new absolute value
+                Volume::Decibels(decibel_offset)
+            };
         }
-    }
-}
 
-impl core::ops::AddAssign<Self> for Volume {
-    fn add_assign(&mut self, rhs: Self) {
-        *self = *self + rhs;
+        let new_db = current_db + decibel_offset;
+        Volume::Decibels(new_db)
     }
-}
-
-impl core::ops::Sub<Self> for Volume {
-    type Output = Self;
-
-    fn sub(self, rhs: Self) -> Self {
-        use Volume::{Decibels, Linear};
 
-        match (self, rhs) {
-            (Linear(a), Linear(b)) => Linear(a - b),
-            (Decibels(a), Decibels(b)) => Decibels(linear_to_decibels(
-                decibels_to_linear(a) - decibels_to_linear(b),
-            )),
-            // {Linear, Decibels} favors the left hand side of the operation by
-            // first converting the right hand side to the same type as the left
-            // hand side and then performing the operation.
-            (Linear(..), Decibels(db)) => self - Linear(decibels_to_linear(db)),
-            (Decibels(..), Linear(l)) => self - Decibels(linear_to_decibels(l)),
-        }
+    /// Increases the volume by the specified percentage.
+    ///
+    /// This method works in the linear domain, where a 100% increase
+    /// means doubling the volume (equivalent to +6.02dB).
+    ///
+    /// # Arguments
+    /// * `percentage` - The percentage to increase (50.0 means 50% increase)
+    ///
+    /// # Examples
+    /// ```
+    /// use bevy_audio::Volume;
+    ///
+    /// let volume = Volume::Linear(1.0);
+    /// let increased = volume.increase_by_percentage(100.0);
+    /// assert_eq!(increased.to_linear(), 2.0);
+    /// ```
+    pub fn increase_by_percentage(&self, percentage: f32) -> Self {
+        let factor = 1.0 + (percentage / 100.0);
+        self.adjust_by_linear(factor)
     }
-}
 
-impl core::ops::SubAssign<Self> for Volume {
-    fn sub_assign(&mut self, rhs: Self) {
-        *self = *self - rhs;
+    /// Decreases the volume by the specified percentage.
+    ///
+    /// This method works in the linear domain, where a 50% decrease
+    /// means halving the volume (equivalent to -6.02dB).
+    ///
+    /// # Arguments
+    /// * `percentage` - The percentage to decrease (50.0 means 50% decrease)
+    ///
+    /// # Examples
+    /// ```
+    /// use bevy_audio::Volume;
+    ///
+    /// let volume = Volume::Linear(1.0);
+    /// let decreased = volume.decrease_by_percentage(50.0);
+    /// assert_eq!(decreased.to_linear(), 0.5);
+    /// ```
+    pub fn decrease_by_percentage(&self, percentage: f32) -> Self {
+        let factor = 1.0 - (percentage / 100.0).clamp(0.0, 1.0);
+        self.adjust_by_linear(factor)
     }
-}
-
-impl core::ops::Mul<Self> for Volume {
```

Comment:
Mul and Div impls didn't need to be removed, they are strictly equivalent to `adjust_by_linear` and `adjust_by_decibel` depending on the variant of the right-hand side